File: Assets/Mycom.Target.Unity/Library/Internal/Implementations/iOS/MyTargetPrivacyProxy.m
#import <MyTargetSDK/MTRGPrivacy.h>

struct MTRGPrivacyValue
{
    int32_t userConsent;
    int32_t ccpaUserConsent;
    int32_t iabUserConsent;
    uint16_t userAgeRestricted;
};

void MTRGSetUserConsent(bool userConsent)
{
    BOOL value = userConsent == true ? YES : NO;
    [MTRGPrivacy setUserConsent:value];
}

void MTRGSetCcpaUserConsent(bool ccpaUserConsent)
{
    BOOL value = ccpaUserConsent == true ? YES : NO;
    [MTRGPrivacy setCcpaUserConsent:value];
}

void MTRGSetIabUserConsent(bool iabUserConsent)
{
    BOOL value = iabUserConsent == true ? YES : NO;
    [MTRGPrivacy setIABUserConsent:value];
}

void MTRGSetUserAgeRestricted(bool userAgeRestricted)
{
    BOOL value = userAgeRestricted == true ? YES : NO;
    [MTRGPrivacy setUserAgeRestricted:value];
}

struct MTRGPrivacyValue MTRGGetCurrentPrivacy()
{
    struct MTRGPrivacyValue result;
    const MTRGPrivacy* currentValue = MTRGPrivacy.currentPrivacy;

    NSNumber* userConsent = [currentValue userConsent];
    result.userConsent = userConsent == nil ? -1 : userConsent.intValue;

    NSNumber* ccpaUserConsent = [currentValue ccpaUserConsent];
    result.ccpaUserConsent = ccpaUserConsent == nil ? -1 : ccpaUserConsent.intValue;

    NSNumber* iabUserConsent = [currentValue iABUserConsent];
    result.iabUserConsent = iabUserConsent == nil ? -1 : iabUserConsent.intValue;

    result.userAgeRestricted = (uint16_t) ([currentValue userAgeRestricted] == YES);

    return result;
}

File: Assets/Mycom.Target.Unity/Library/Internal/MyTargetPrivacyProxy.cs
namespace Mycom.Target.Unity.Internal
{
    using System;

    internal static partial class MyTargetPrivacyProxy
    {
        internal static Boolean UserConsent
        {
            set
            {
                try
                {
                    SetUserConsent(value);
                }
                catch (Exception ex)
                {
                    MyTargetLogger.Log(ex.ToString());
                }
            }
        }

        internal static Boolean CcpaUserConsent
        {
            set
            {
                try
                {
                    SetCcpaUserConsent(value);
                }
                catch (Exception ex)
                {
                    MyTargetLogger.Log(ex.ToString());
                }
            }
        }

        internal static Boolean IabUserConsent
        {
            set
            {
                try
                {
                    SetIabUserConsent(value);
                }
                catch (Exception ex)
                {
                    MyTargetLogger.Log(ex.ToString());
                }
            }
        }

        internal static Boolean UserAgeRestricted
        {
            set
            {
                try
                {
                    SetUserAgeRestricted(value);
                }
                catch (Exception ex)
                {
                    MyTargetLogger.Log(ex.ToString());
                }
            }
        }

        internal static MyTargetPrivacyValue CurrentPrivacy
        {
            get
            {
                MyTargetPrivacyValue result = new MyTargetPrivacyValue();
                try
                {
                    GetCurrentPrivacy(ref result);
                }
                catch (Exception ex)
                {
                    MyTargetLogger.Log(ex.ToString());
                }
                return result;
            }
        }

        static partial void SetUserConsent(Boolean userConsent);

        static partial void SetCcpaUserConsent(Boolean ccpaUserConsent);

        static partial void SetIabUserConsent(Boolean iabUserConsent);

        static partial void SetUserAgeRestricted(Boolean userAgeRestricted);

        static partial void GetCurrentPrivacy(ref MyTargetPrivacyValue value);

        internal struct MyTargetPrivacyValue
        {
            internal Boolean? userConsent;
            internal Boolean? ccpaUserConsent;
            internal Boolean? iabUserConsent;
            internal Boolean userAgeRestricted;
        }
    }

#if !UNITY_IOS && !UNITY_ANDROID
    internal static partial class MyTargetPrivacyProxy
    {
        static partial void SetUserConsent(Boolean userConsent) { }

        static partial void SetCcpaUserConsent(Boolean ccpaUserConsent) { }

        static partial void SetIabUserConsent(Boolean iabUserConsent) { }

        static partial void SetUserAgeRestricted(Boolean userAgeRestricted) { }

        static partial void GetCurrentPrivacy(ref MyTargetPrivacyValue value) { }
    }
#endif

}